From 942151a4e8594a79037826963a5bc1e1add7bc3b Mon Sep 17 00:00:00 2001
From: Ozkan Sezer <sezeroz@gmail.com>
Date: Thu, 13 Mar 2025 04:04:10 +0300
Subject: [PATCH] music_mpg123.c: disable _FILE_OFFSET_BITS suffixes on
 libmpg123 calls.

Avoids libmpg123's terrifying mess. Useful if mpg123_shared is disabled
and if SDL_mixer is built with _FILE_OFFSET_BITS=64.

(cherry picked from commit 72a73339731a12c1002f9caca64f1ab924938102)
---
 src/codecs/music_mpg123.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/codecs/music_mpg123.c b/src/codecs/music_mpg123.c
index 90750aeb8..4457250ef 100644
--- a/src/codecs/music_mpg123.c
+++ b/src/codecs/music_mpg123.c
@@ -30,6 +30,7 @@
 #include "mp3utils.h"
 
 #include <stdio.h>      /* For SEEK_SET */
+#define MPG123_NO_LARGENAME /* disable the _FILE_OFFSET_BITS suffixes. */
 #ifdef MPG123_HEADER
 #include MPG123_HEADER
 #else
